### ISSUE TITLE
Errors in Radicale console window

### ISSUE THREAD

**Issue by pierredGitHub** [Kozea/Radicale#846]
I am using Radicale 2.1.9 with Python 3.6 under Windows 10. I am using it to sync contacts between 2 computers and 2 phones. It's working fine as far as I can tell, but I always see the same error multiple times in the console window on the computer where Radicale is running. The error is:

[xxxx] WARNING: Bad PUT request on 'long_filename.vcf': Failed to store item 'long_filename.vcf' in collection 'long_collection_name': a bytes-like object is required, not 'str'

This error is repeated multiple times for different .vcf files. When I open any .vcf file (with Notepad), they look fine to me. What is causing these errors?

I have googled for it and always find results that talk about fixing the code. Well, I didn't write the code. I am just an end user. Is this a bug in Radicale?

Any ideas on a workaround or fix? Maybe a configuration setting?

Thanks.

Pierre

**Comment by pierredGitHub** [comment]
Hi m4tze,

Yes indeed many of my contacts have pictures. That would definitely explain I get the error so many times for so many different .vcf files. But the contacts are not changing that often so not sure why I'd get the errors in the console window so often.

In any case, the pictures seem to sync just fine.

So ..... I have looked at the other thread you referenced (#721). I see that replacing BASE64 with b appears to be a fix. But where do I do that? I suspect it's in the code, so that's probably not for me. But can the fix be brought in by a new version of Radicale? Or is the issue in Python? Or something else?

At least now I know what's causing it, even if I can't fix it on my own. Thanks for that.

**Comment by m4tze** [comment]
Hi pierredGitHub,

your vcf file has plain text content. This can be changed with every text editor on your system. Thus, you might "search and replace" the Term 'BASE64' with 'b' in your vcf file(s) on your own before uploading them to radicale. So no code change needed, just a change on the data side.

- Regards

**Comment by pierredGitHub** [comment]
I am aware that vcf files are plain text. I am using Radicale to sync between 2 phones and a computer. On the computer, it's Thunderbird, using the CardBook add-on. No VCF files there that I can find. On the phones (BlackBerry and iPhone) no VCF files that I am aware of. So that leaves the Radicale collection. In the collection, all the vcf files that have pictures have "ENCODING=B" in them. None of them have "ENCODING=BASE64" (I did a search in the folder). So I am at a loss here. Am I looking in the wrong place?

**Comment by io-error** [comment]
Hi pierredGitHub, 

are your contacts in vCard version 3 or 4? Yesterday I had the same issue with vCard4 contacts with a photo and I guess Radicale doesn‘t support vCard4 since vobject doesn‘t support it either (https://github.com/eventable/vobject/issues/106). So I assume that vobject and therfore also Radicale can‘t handle the changed photo description of vCard4 correctly.

V3.0: PHOTO;TYPE=JPEG;ENCODING=b:[base64-data] 
V4.0: PHOTO:data:image/jpeg;base64,[base64-data]

**Comment by pierredGitHub** [comment]
Sorry for taking so long to respond. After I finally switched my main phone from the BlackBerry to the iPhone, I deleted the picture for the offending contacts (3 or 4 of them, even though I have plenty more contacts with a picture). Then I let Radicale sync in every direction with all the clients, and then re-added the picture (from the iPhone or Thunderbird, I can't recall now). It resynced fine (including the BlackBerry, and those errors have not reappeared since.
I think the issue was the size of the picture (~1MB or more), but I am not sure about that.
By the way, some of my VCARDS are 4.0 while the vast majority are 3.0 (27 out of 622). I haven't figured out a pattern though for the 4.0 ones.
